pkg/info/info.go

```diff
@@ -0,0 +1,34 @@
+package info
+
+import (
+	"context"
+
+	"github.com/zitadel/oidc/v3/pkg/oidc"
+)
+
+type providerInfoKey struct{}
+
+type ProviderInfo struct {
+	IntrospectionResponse *oidc.IntrospectionResponse
```

Comment:
i wouldn't tie the info to the `*oidc.introspectionResponse` first of all that can be generic and what about userinfo, non oidc related info, ...?

Reply:
That's why the response is part of the struct, if we have other or additional information wie want to add to the context we either can split it or for the moment add it here with a set-function